server/lease/lessor.go

```diff
@@ -446,6 +446,7 @@ func (le *lessor) Promote(extend time.Duration) {
 		l.refresh(extend)
 		item := &LeaseWithTime{id: l.ID, time: l.expiry}
 		le.leaseExpiredNotifier.RegisterOrUpdate(item)
+		le.scheduleCheckpointIfNeeded(l)
```

Comment:
Found that commenting out this line doesn't break the tests. Can you add a test for this?

Reply:
Yes. I wonder why `le.scheduleCheckpointIfNeeded(l)' in line 487 is not handling this case.